packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts

```diff
@@ -54,15 +68,174 @@ export class PluginApiFrontendContribution implements CommandContribution {
+            execute: () => open(this.openerService, createUntitledResource().uri)
+        });
+        commands.registerCommand({ id: 'workbench.action.files.openFile' }, {
+            execute: () => commands.executeCommand(WorkspaceCommands.OPEN_FILE.id)
+        });
+        commands.registerCommand({ id: 'workbench.action.files.openFolder' }, {
+            execute: () => commands.executeCommand(WorkspaceCommands.OPEN_FOLDER.id)
+        });
+        commands.registerCommand({ id: 'workbench.action.files.save', }, {
+            execute: (uri?: monaco.Uri) => {
+                if (uri) {
+                    const uriString = uri.toString();
+                    const widget = this.shell.widgets.find(w => {
+                        const resourceUri = Saveable.is(w) && NavigatableWidget.is(w) && w.getResourceUri();
+                        return (resourceUri && resourceUri.toString()) === uriString;
+                    });
+                    if (Saveable.is(widget)) {
+                        Saveable.save(widget);
+                    }
+                } else {
+                    this.shell.save();
+                }
+            }
+        });
+        commands.registerCommand({ id: 'workbench.action.files.saveAll', }, {
+            execute: () => this.shell.saveAll()
+        });
+        commands.registerCommand({ id: 'workbench.action.closeActiveEditor' }, {
+            execute: (uri?: monaco.Uri) => {
+                let widget = this.editors.currentEditor || this.shell.currentWidget;
+                if (uri) {
+                    const uriString = uri.toString();
+                    widget = this.shell.widgets.find(w => {
+                        const resourceUri = NavigatableWidget.is(w) && w.getResourceUri();
+                        return (resourceUri && resourceUri.toString()) === uriString;
+                    });
+                }
+                if (NavigatableWidget.is(widget)) {
+                    widget.close();
+                }
+            }
+        });
+        commands.registerCommand({ id: 'workbench.action.closeOtherEditors' }, {
+            execute: (uri?: monaco.Uri) => {
+                let editor = this.editors.currentEditor || this.shell.currentWidget;
+                if (uri) {
+                    const uriString = uri.toString();
+                    editor = this.editors.all.find(e => {
+                        const resourceUri = e.getResourceUri();
+                        return (resourceUri && resourceUri.toString()) === uriString;
+                    });
+                }
+                for (const widget of this.shell.widgets) {
+                    if (NavigatableWidget.is(widget) && widget !== editor) {
+                        widget.close();
+                    }
+                }
+            }
+        });
+        commands.registerCommand({ id: 'workbench.action.closeEditorsInGroup' }, {
```

Comment:
should these commands around editors be provided by the core instead ? it looks they're not only targeting plug-in system but can be useful for anyone ?

Reply:
I don't want to mix them:
- `closeEditorsInGroup` would be mean in Theia means different compare to VS Code, in VS Code everything is main area is an editor (webviews, previews and so on), in our case editor is a text editor only
- this commands accept arguments which don't have representation in Theia, like view positions and so on, for now i just did not declare them (so even implementation is not really complete), but it does not mean VS Code extensions cannot pass them

Reply:
We have close tab command already and you can see that it does completely another thing.